src/components/dateTimePickerInput.js

```diff
@@ -28,287 +28,259 @@
         }
-      } else {
-        setSelectedDate(null);
+        // convert to slashes because it conflicts with the MUI DateTimeCmp
+        const parsedValueWithSlashes = dateText.replace(/-/g, '/');
+        return new Date(parsedValueWithSlashes);
       }
-    }, [parsedValue]);
+      return '';
+    }
 
-    B.defineFunction('Clear', () => setSelectedDate(null));
-    B.defineFunction('Enable', () => setIsDisabled(false));
-    B.defineFunction('Disable', () => setIsDisabled(true));
+    function isValidDate(date) {
+      return date instanceof Date && !isNaN(date);
+    }
 
     let DateTimeComponent;
-    let format;
-    let resultString;
-    let use24HourClock = true;
-    let minDate;
-    let maxDate;
-
-    switch (type) {
+    let resultValue;
+    switch (typeComponent) {
       case 'date': {
         DateTimeComponent = KeyboardDatePicker;
-        format = dateFormat || 'dd/MM/yyyy';
-
-        minDate = convertToValidDate(minValueText);
-        maxDate = convertToValidDate(maxValueText);
-
-        resultString = isValidDate(selectedDate)
+        resultValue = isValidDate(selectedDate)
           ? DateFns.format(selectedDate, 'yyyy-MM-dd')
           : null;
         break;
       }
       case 'datetime': {
         DateTimeComponent = KeyboardDateTimePicker;
-        format = datetimeFormat || 'dd/MM/yyyy HH:mm:ss';
-        use24HourClock = use24HourClockTime;
-
-        resultString = isValidDate(selectedDate)
+        resultValue = isValidDate(selectedDate)
           ? selectedDate.toISOString()
           : null;
+
         break;
       }
       case 'time': {
         DateTimeComponent = KeyboardTimePicker;
-        format = timeFormat || 'HH:mm:ss';
-        use24HourClock = use24HourClockTime;
-
-        resultString = isValidDate(selectedDate)
+        resultValue = isValidDate(selectedDate)
           ? DateFns.format(selectedDate, 'HH:mm:ss')
           : null;
         break;
       }
       default:
     }
 
-    const onBlurHandler = (event) => {
-      const {
-        target: { validity },
-      } = event;
-
-      handleValidation(validity);
-      setAfterFirstValidation(true);
-    };
-
     const DateTimeCmp = (
       <DateTimeComponent
         id={labelControlRef}
-        error={errorState}
-        value={selectedDate}
-        size={size}
-        onBlur={onBlurHandler}
-        autoComplete={autoComplete ? 'on' : 'off'}
         classes={{
           root: `${classes.formControl} ${floatLabel && classes.floatLabel}`,
         }}
-        variant={variant}
-        placeholder={placeholderText}
+        value={selectedDate}
+        autoComplete={autoComplete ? 'on' : 'off'}
+        placeholder={useText(placeholder)}
+        label={!hideLabel && useText(label)}
+        error={errorState}
+        helperText={currentHelperText}
+        disableToolbar={disableToolbar}
+        disablePast={disablePastDates}
+        minDate={convertToValidDate(useText(minValue))}
```

Comment:
Min date doesn't work when I test it. I expect to see a message before I submit the form.
My option value = 01-10-2024
Runtime value = 01-09-2024